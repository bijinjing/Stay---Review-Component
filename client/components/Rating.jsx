import React from 'react';
import Ratings from 'react-ratings-declarative';
import styled from 'styled-components'


const Rating_section = styled.div`
  margin-top:10px
  margin-bottom:10px
`;

const Rating_column = styled.div`
  display:inline-flex
  width:50%
  position:relative
`;

const Rating_column_description_1 = styled.div`
  display:block
  text-align: left;
`;

const Rating_column_description_2 = styled.div`
  display:block
  text-align: left;
`;

const Rating_column_rating_1 = styled.div`
  display:block;
  align: right;
  margin-left:10%
`;

const Rating_column_rating_2 = styled.div`
  display:block;
  align: right;
  margin-left:25%
`;

const Rating_category = styled.div`
  line-height: 1.75em
  font-weight: 330
  font-size: 16px
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif
  word-wrap: break-word
  boarder:5px
`;

const Rating_star = styled.div`
  width: 110px;
  margin:6px
`;


const NewRating = ({newRating})=>{
  return(
    <Ratings
    rating={newRating}
    widgetRatedColors="#008489"
    widgetDimensions="17px"
    widgetSpacings='1px'
   >
    <Ratings.Widget/>
    <Ratings.Widget/>
    <Ratings.Widget/>
    <Ratings.Widget/>
    <Ratings.Widget/>
   </Ratings>
  )
}


const RatingComp = ({rating}) => {
  return(
    <Rating_section>
      <Rating_column>
        <Rating_column_description_1>
          <Rating_category>Accuracy </Rating_category>
          <Rating_category>Communication</Rating_category>
          <Rating_category>Cleanliness </Rating_category>
        </Rating_column_description_1>
        <Rating_column_rating_1>
          <Rating_star><NewRating newRating={rating.rating_accuracy}/></Rating_star>
          <Rating_star><NewRating newRating={rating.rating_communication}/></Rating_star>
          <Rating_star><NewRating newRating={rating.rating_cleanliness}/></Rating_star>
        </Rating_column_rating_1>
      </Rating_column>
      <Rating_column>
        <Rating_column_description_2>
          <Rating_category>Location </Rating_category>
          <Rating_category>Check-in </Rating_category>
          <Rating_category>Value </Rating_category>
        </Rating_column_description_2>
        <Rating_column_rating_2>
          <Rating_star><NewRating newRating={rating.rating_location}/></Rating_star>
          <Rating_star><NewRating newRating={rating.rating_checkin}/></Rating_star>
          <Rating_star><NewRating newRating={rating.rating_value}/></Rating_star>
        </Rating_column_rating_2>
      </Rating_column>
    </Rating_section>
  )
}

export default RatingComp