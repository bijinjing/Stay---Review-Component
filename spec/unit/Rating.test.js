import React from 'react';
import ReactDOM from 'react-dom';
import Rating from '../../client/components/Rating.jsx';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme'
Enzyme.configure({ adapter: new Adapter() });

describe('Rating', ()=>{
  it('renders without crashing', () => {
    const props = {
      rating:{
        rating_accuracy:5,
        rating_communication:4.5,
        rating_cleanliness:4,
        rating_location:5,
        rating_checkin:4.5,
        rating_value:4
      }
    }
    const wrapper = shallow(<Rating {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})