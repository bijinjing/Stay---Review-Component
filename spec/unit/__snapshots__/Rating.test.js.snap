// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders without crashing 1`] = `
<Rating__Rating_section>
  <Rating__Rating_column>
    <Rating__Rating_column_description_1>
      <Rating__Rating_category>
        Accuracy 
      </Rating__Rating_category>
      <Rating__Rating_category>
        Communication
      </Rating__Rating_category>
      <Rating__Rating_category>
        Cleanliness 
      </Rating__Rating_category>
    </Rating__Rating_column_description_1>
    <Rating__Rating_column_rating_1>
      <Rating__Rating_star>
        <NewRating
          newRating={5}
        />
      </Rating__Rating_star>
      <Rating__Rating_star>
        <NewRating
          newRating={4.5}
        />
      </Rating__Rating_star>
      <Rating__Rating_star>
        <NewRating
          newRating={4}
        />
      </Rating__Rating_star>
    </Rating__Rating_column_rating_1>
  </Rating__Rating_column>
  <Rating__Rating_column>
    <Rating__Rating_column_description_2>
      <Rating__Rating_category>
        Location 
      </Rating__Rating_category>
      <Rating__Rating_category>
        Check-in 
      </Rating__Rating_category>
      <Rating__Rating_category>
        Value 
      </Rating__Rating_category>
    </Rating__Rating_column_description_2>
    <Rating__Rating_column_rating_2>
      <Rating__Rating_star>
        <NewRating
          newRating={5}
        />
      </Rating__Rating_star>
      <Rating__Rating_star>
        <NewRating
          newRating={4.5}
        />
      </Rating__Rating_star>
      <Rating__Rating_star>
        <NewRating
          newRating={4}
        />
      </Rating__Rating_star>
    </Rating__Rating_column_rating_2>
  </Rating__Rating_column>
</Rating__Rating_section>
`;
